xhttp/request_metrics.go

```diff
@@ -5,22 +5,25 @@ import (
 	"strconv"
 	"time"
 
+	"github.com/go-phorce/dolly/algorithms/math"
 	"github.com/go-phorce/dolly/metrics"
 	"github.com/go-phorce/dolly/metrics/tags"
 	"github.com/go-phorce/dolly/xhttp/identity"
 )
 
 // a http.Handler that records execution metrics of the wrapper handler
 type requestMetrics struct {
-	handler       http.Handler
-	responseCodes []string
+	handler           http.Handler
+	responseCodes     []string
+	additionalMetrics map[string]string
 }
 
 // NewRequestMetrics creates a wrapper handler to produce metrics for each request
-func NewRequestMetrics(h http.Handler) http.Handler {
+func NewRequestMetrics(h http.Handler, additionalMetrics map[string]string) http.Handler {
```

Comment:
please keep this `NewRequestMetrics` with unmodified parameters and create 
`NewRequestWithExtraMetrics` , than `NewRequestMetrics` forwards to `NewRequestWithExtraMetrics` with `nil`